velox/exec/tests/utils/PlanBuilder.cpp

```diff
@@ -716,6 +716,57 @@ PlanBuilder& PlanBuilder::localMerge(const std::vector<std::string>& keys) {
   return *this;
 }
 
+PlanBuilder& PlanBuilder::expand(
+    const std::vector<std::vector<std::string>>& projectionSets) {
+  std::vector<std::vector<core::TypedExprPtr>> projectSetExprs;
+  projectSetExprs.reserve(projectionSets.size());
+  std::vector<std::string> names;
+  names.reserve(projectionSets[0].size());
+  std::vector<std::shared_ptr<const Type>> types;
+  types.reserve(projectionSets[0].size());
+  std::string groupIdPrefix = "group_id_";
+  int grouIdColCount = 0;
+  for (int i = 0; i < projectionSets[0].size(); ++i) {
+    for (int j = 0; j < projectionSets.size(); ++j) {
+      if (projectionSets[j][i] != "") {
+        if (planNode_->outputType()->containsChild(projectionSets[j][i])) {
+          names.push_back(projectionSets[j][i]);
+          types.push_back(
+              field(planNode_->outputType(), projectionSets[j][i])->type());
+        } else {
```

Comment:
It looks like 'projection' can refer to non-existent column. Let's not allow this and fail fast.

Reply:
Yes, in this case, the group_id constant column does not exist in the input row type. It is generated within the Expand operator and then passed to the subsequent Aggregate operator. Eventually, it will be removed in the final result.

Reply:
I'm confused. I didn't think that Expand operator computes anything on its own, just uses specified columns and constants to generate results. Is this not the case? Please, clarify.

Reply:
@mbasmanova  Based on the provided [example](https://github.com/facebookincubator/velox/pull/5403#discussion_r1328159472):
For the constant columns group_id_0 and group_id_1, it's important to note that these two columns are not part of the input row type. In the PlanBuild process, we generate the name and type for these constant columns. To clarify, the generation of these constant columns occurs in the PlanBuild phase, not within the Expand operation. The Expand operator utilizes the specified columns and constants to generate the desired results.

Reply:
Got it. Thank you for clarifying.

I wonder if a cleaner API would be for projections to be SQL expressions: 

```
- c0
- 123
- null
- null:varchar
- null as 'foo'
```

This will allow the caller to specify the type for null and non-null constants as well as specify columns names.

Reply:
@mbasmanova Currently, we represent null columns with "". Are you suggesting changing "" to null, null:varchar, null as 'foo'?

Reply:
Yes, I suggest using SQL expression instead of an empty string. This allows to specify type and column name and is consistent with other PlanBuilder APIs.

Reply:
Please, address this comment.